src/Backends/REFPROP/REFPROPMixtureBackend.cpp

```diff
@@ -1218,6 +1271,244 @@ phases REFPROPMixtureBackend::GetRPphase()
+	hL_mol = calc_saturated_liquid_keyed_output(iHmolar);
+       	hV_mol = calc_saturated_vapor_keyed_output(iHmolar);
+  
+	// Derivative of temperature w.r.t. pressure ALONG the saturation curve
+	CoolPropDbl dTdP_sat = _T*(1/rhoV_mol - 1/rhoL_mol)/(hV_mol - hL_mol);
+
+	// "Trivial" inputs
+	if (Of1 == iT && Wrt1 == iP){ return dTdP_sat;}
+	else if (Of1 == iP && Wrt1 == iT){ return 1/dTdP_sat;}
+	// Derivative taken with respect to T
+	else if (Wrt1 == iT){
+		return first_partial_deriv(Of1, iT, iP) + first_partial_deriv(Of1, iP, iT)/dTdP_sat;
+	}
+	// Derivative taken with respect to p
+	else if (Wrt1 == iP){
+		return first_partial_deriv(Of1, iP, iT) + first_partial_deriv(Of1, iT, iP)*dTdP_sat;
+	}
+	else{
+		throw ValueError(format("Not possible to take first saturation derivative with respect to %s", get_parameter_information(Wrt1,"short").c_str()));
+	}
+}
+
+CoolPropDbl REFPROPMixtureBackend::calc_first_saturation_deriv(parameters Of1, parameters Wrt1, CoolPropDbl rhoL_mol, CoolPropDbl rhoV_mol) {
+        this->check_loaded_fluid();
+	
+        if (_RPcheckTwophase() == false)
+	  throw ValueError("Not two phase fluid: saturation derivative is not possible");
+
+	double hL_mol, hV_mol;
+	hL_mol = calc_saturated_liquid_keyed_output(iHmolar);
+       	hV_mol = calc_saturated_vapor_keyed_output(iHmolar);
+	
+	// Derivative of temperature w.r.t. pressure ALONG the saturation curve
+	CoolPropDbl dTdP_sat = _T*(1/rhoV_mol - 1/rhoL_mol)/(hV_mol - hL_mol);
+
+	// "Trivial" inputs
+	if (Of1 == iT && Wrt1 == iP){ return dTdP_sat;}
+	else if (Of1 == iP && Wrt1 == iT){ return 1/dTdP_sat;}
+	// Derivative taken with respect to T
+	else if (Wrt1 == iT){
+		return first_partial_deriv(Of1, iT, iP) + first_partial_deriv(Of1, iP, iT)/dTdP_sat;
+	}
+	// Derivative taken with respect to p
+	else if (Wrt1 == iP){
+		return first_partial_deriv(Of1, iP, iT) + first_partial_deriv(Of1, iT, iP)*dTdP_sat;
+	}
+	else{
+		throw ValueError(format("Not possible to take first saturation derivative with respect to %s", get_parameter_information(Wrt1,"short").c_str()));
+	}
+}
+
+CoolPropDbl REFPROPMixtureBackend::calc_first_two_phase_deriv(parameters Of, parameters Wrt, parameters Constant) {
+       this->check_loaded_fluid();
+	
+       if (_RPcheckTwophase() == false)
+	 throw ValueError("Not two phase fluid: saturation derivative is not possible");
+    
+       shared_ptr<REFPROPMixtureBackend> SatL(new REFPROPMixtureBackend(this->calc_fluid_names())), SatV(new REFPROPMixtureBackend(this->calc_fluid_names()));
+       SatL->set_mole_fractions(this->get_mole_fractions());
+       SatV->set_mole_fractions(this->get_mole_fractions());
+       SatL->update(QT_INPUTS, 0.0, _T); // <--- This looks never effecient,
+       SatV->update(QT_INPUTS, 1.0, _T); //      but just mimics the saturate states of HEOS..
+       
+       double rhoV_mol = _rhoVmolar, rhoL_mol = _rhoVmolar; //This looks redundant, but, by this, we avoid annoying static casts ..
```

Comment:
```suggestion
       double rhoV_mol = _rhoVmolar, rhoL_mol = _rhoLmolar; //This looks redundant, but, by this, we avoid annoying static casts ..
```
that's a BIG mistake!